packages/web-components/src/player-component/player.class.ts

```diff
@@ -206,6 +206,14 @@ export class PlayerWrapper {
         this._liveStream = value;
     }
 
+    public get liveStream() {
+        const url = new URL(this._liveStream);
+        if (this.accessToken) {
+            url.searchParams.set('authorization', this.accessToken);
```

Comment:
There needs to be an equivalent way for Portal to set the rtsp URL to rtsp://localhost (let's say, this.rtspUrl), so that if (this.rtspUrl) url.searchParams.set('rtsp', encodeURIComponent(this.rtslUrl));

Reply:
Compromise suggestion: add a new optional member to ISource, say, rtspTunnelUrl?: string, and if null or empty (be sure to explicitly check for empty) then url.searchParams.set('rtsp', encodeURIComponent(this.rtspTunnelUrl && this.rtspTunnelUrl.length > 0 ? this.rtspTunnelUrl : "rtsp://localhost"));. In this way, when Portal starts passing non-empty rtspTunnelUrl, this code won't have to change in future.

Reply:
Talked to Girish and Raymond.Making a change to consolidate the code to one place in player.class.ts for now and eventually move it out to video-analyzer-player  post ignite.